graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+        df = self.df.loc[self.df.experiment_type == type]
+
+        if update:
+            self.df = df
+        return df
+
+    def longer_than(self, length: int, update: bool = False) -> pd.DataFrame:
+        """Select molecules longer than a given length.
+
+        :param length: Minimum length of molecule.
+        :type length: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.length > length]
+
+        if update:
+            self.df = df
+        return df
+
+    def shorter_than(self, length: int, update: bool = False) -> pd.DataFrame:
+        """
+        Select molecules shorter than a given length.
+
+        :param length: Maximum length of molecule.
+        :type length: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.length < length]
+
+        if update:
+            self.df = df
+        return df
+
+    def resolution_higher_than_or_equal_to(
+        self, resolution: int, update: bool = False
+    ) -> pd.DataFrame:
+        """Select molecules with a resolution higher than or equal to the given value.
+
+        Conventions for PDB resolution values are used, where a lower resolution
+        value indicates a higher resolution for a molecule overall.
+
+        :param resolution: Minimum resolution of molecule.
+        :type resolution: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.resolution <= resolution]
+
+        if update:
+            self.df = df
+        return df
+
+    def resolution_lower_than_or_equal_to(
+        self, resolution: int, update: bool = False
+    ) -> pd.DataFrame:
+        """Select molecules with a resolution lower than or equal to the given value.
+
+        Conventions for PDB resolution values are used, where a higher resolution
+        value indicates a lower resolution for a molecule overall.
+
+        :param resolution: Maximum resolution of molecule.
+        :type resolution: int
+        :param update: Whether to modify the DataFrame in place, defaults to
+            ``False``.
+        :type update: bool, optional
+
+        :return: DataFrame of selected molecules.
+        :rtype: pd.DataFrame
+        """
+        df = self.df.loc[self.df.resolution >= resolution]
+
+        if update:
+            self.df = df
+        return df
+
+    def oligomeric(self, oligomer: int = 1, update: bool = False):
```

Comment:
I noticed a bug here. I made a method like this:

```python
    def oligomeric(self, oligomer: int = 1, comparison: str = "equal", update: bool = False):
        """
        Select molecules with a given oligmeric length.

        :param length: Oligomeric length of molecule, defaults to ``1``.
        :type length: int
        :param comparison: Comparison operator. One of ``"equal"``,
            ``"less"``, or ``"greater"`` defaults to ``"equal"``.
        :param update: Whether to modify the DataFrame in place, defaults to
            ``False``.
        :type update: bool, optional
        :return: DataFrame of selected oligmers.
        :rtype: pd.DataFrame
        """
        if comparison == "equal":
            df = self.df[
                self.df.groupby("pdb")["pdb"].transform("size") == oligomer
            ]
        elif comparison == "less":
            df = self.df[
                self.df.groupby("pdb")["pdb"].transform("size") < oligomer
            ]
        elif comparison == "greater":
            df = self.df[
                self.df.groupby("pdb")["pdb"].transform("size") > oligomer
            ]
        else:
            raise ValueError(
                "Comparison must be one of 'equal', 'less', or 'greater'."
            )

        if update:
            self.df = df
        return df
```

What do you think about using the `comparison` pattern throughout (instead of having multiple methods for each)

Reply:
I guess it only suits `longer_than` and `shorter_than`. I was also thinking about the resolution methods prior to the syntax change.

Reply:
I've made this "length checking" functionality more generic in [this commit](https://github.com/a-r-j/graphein/pull/272/commits/7a6ff724638837fb3ec1509b12ecae91e8a543da).